lib/panko/serializer_resolver.rb

```diff
@@ -1,25 +1,35 @@
 # frozen_string_literal: true
 
 class Panko::SerializerResolver
-  def self.resolve(name)
-    serializer_name = "#{name.singularize.camelize}Serializer"
-    serializer_const = safe_const_get(serializer_name)
+  class << self
+    def resolve(name, from)
+      serializer_const = nil
 
-    return nil if serializer_const.nil?
-    return nil unless is_serializer(serializer_const)
+      if namespace = namespace_for(from)
+        serializer_const = safe_serializer_get("#{namespace}::#{name.singularize.camelize}Serializer")
+      end
 
-    serializer_const
-  end
+      serializer_const ||= safe_serializer_get("#{name.singularize.camelize}Serializer")
+      serializer_const
+    end
 
-  private
+    private
 
-  def self.is_serializer(const)
-    const < Panko::Serializer
-  end
+    if Module.method_defined?(:module_parent_name)
```

Comment:
Rails 6.0 renamed `Module#parent_name` into `Module#module_parent_name`, so to support Rails 5.2 we need to do some feature checking.